minorminer/utils/zephyr/node_edge.py

```diff
@@ -0,0 +1,635 @@
+        if not isinstance(new_ccoord, CartesianCoord):
+            raise TypeError(
+                f"Expected ccoord to be CartesianCoord or tuple[int], got {type(new_ccoord)}"
+            )
+        for c in (new_ccoord.x, new_ccoord.y):
+            if not isinstance(c, int):
+                raise TypeError(f"Expected ccoord.x and ccoord.y to be 'int', got {type(c)}")
+            if c < 0:
+                raise ValueError(f"Expected ccoord.x and ccoord.y to be non-negative, got {c}")
+        if new_ccoord.x % 2 == new_ccoord.y % 2:
+            raise ValueError(
+                f"Expected ccoord.x and ccoord.y to differ in parity, got {new_ccoord.x, new_ccoord.y}"
+            )
+        # check k value of CartesianCoord is consistent with t
+        if hasattr(self, "_shape"):
+            if (self._shape.t is None) != (new_ccoord.k is None):
+                raise ValueError(
+                    f"shape and ccoord must be both quotient or non-quotient, got {self._shape}, {new_ccoord}"
+                )
+            if (self._shape.t is not None) and (new_ccoord.k not in range(self._shape.t)):
+                raise ValueError(f"Expected k to be in {range(self._shape.t)}, got {new_ccoord.k}")
+
+            # check x, y value of CartesianCoord is consistent with m
+            if self._shape.m is not None:
+                if all(val in range(4 * self._shape.m + 1) for val in (new_ccoord.x, new_ccoord.y)):
+                    self._ccoord = new_ccoord
+                else:
+                    raise ValueError(
+                        f"Expected ccoord.x and ccoord.y to be in {range(4*self._shape.m+1)}, got {new_ccoord.x, new_ccoord.y}"
+                    )
+        self._ccoord = new_ccoord
+
+    @staticmethod
+    def get_coord(coord: tuple[int]) -> CartesianCoord | ZephyrCoord:
+        """Takes a tuple[int] and returns the corresponding CartesianCoord
+        or ZephyrCoord"""
+        if (not isinstance(coord, tuple)) or (not all(isinstance(c, int) for c in coord)):
+            raise TypeError(f"Expected {coord} to be a tuple[int], got {coord}")
+        if any(c < 0 for c in coord):
+            raise ValueError(f"Expected elements of coord to be non-negative, got {coord}")
+        len_coord = len(coord)
+        if len_coord in (2, 3):
+            x, y, *k = coord
+            if x % 2 == y % 2:
+                raise ValueError(f"Expected x, y to differ in parity, got {x, y}")
+            return CartesianCoord(x=x, y=y) if len(k) == 0 else CartesianCoord(x=x, y=y, k=k[0])
+        if len_coord in (4, 5):
+            u, w, *k, j, z = coord
+            for var, val in {"u": u, "j": j}.items():
+                if not val in [0, 1]:
+                    raise ValueError(f"Expected {var} to be in [0, 1], got {val}")
+            return (
+                ZephyrCoord(u=u, w=w, j=j, z=z)
+                if len(k) == 0
+                else ZephyrCoord(u=u, w=w, k=k[0], j=j, z=z)
+            )
```

Comment:
I usually prefer _not_ using the `... if ... else ...` and instead write it out:
```suggestion
            if len(k) == 0
                return ZephyrCoord(u=u, w=w, j=j, z=z)
            else 
                return ZephyrCoord(u=u, w=w, k=k[0], j=j, z=z)
```

Reply:
Done in https://github.com/dwavesystems/minorminer/pull/261/commits/70d84ed616e04331a0f181015fdf1fb3e2774596